design/button.behavior.md

```diff
@@ -0,0 +1,94 @@
+# Button Behavior
+
+Button behavior is the high level representation of how an MRE actor should behave as a button.  This behavior consists of two actions
+that occur on the object when it is targeted and when it is activated or clicked.  As part of these interactions, the point of targeting
+of the button will be passed along as additional data to the app for consumption.
+
+Button Behavior Actions:
+- `hover` - The action that occurs when the button is targeted by the host app input system. 
+    - Hover states: `enter`, `hovering`, `exit`
+- `click` - The action that occurs when a complete click has happened on the button consisting of both a button pressed and button released event. 
+    - Click states: None.  This is a single atomic event.
+- `button` - The action that occurs when the button interaction state has changed from being pressed and then released.
+    - Button states: `pressed`, `holding`, `released`
+    
+Button Event Data:
+- `targetedPoints` - The collection of target points for the current state event of the action.  This collection is of `PointData` that contains both app space and local target object space versions of the points.
+
+Button Event Data Per Action:
+- Hover Action:
+	- `enter` - single app coordinate target point for where the hover is first entered.
+	- `hovering` - collection of app coordinate target points for the time while hovering is occuring, sent at a set frequency.
+	- `exit` - single app coordinate target point for where the hover was exited.
+- Click Action:
+	- There is only the click state and will contain the single app coordinate target point of the click action.
+- Button Action:
+	- `pressed` - single app coordinate target point for where the pressed event began.
+	- `holding` - collection of app coordinate target points for the time while the button is held down, sent at a set frequency.
+	- `released` - single app coordinate target point for where the button is released.
+	
+## Architecture
+
+### PointData
+``` ts
+/**
+ * Interfaace that represents a point in space as a local space point and an app space point.
+ */
+export interface PointData {
+	/**
+	 * The app coordinate space target points collected for the event
+	 */
+	appSpacePoint: Vector3Like;
+
+	/**
+	 * The local coordinate space target points collected for the event.
+	 */
+	localSpacePoint: Vector3Like;
```

Comment:
objectSpacePoint

Reply:
All of our vocabulary around local versus app space coordinates involve either the point being in app or local space.  I hesitate to change that without a global shift to that vocabulary.

---
In reply to: [428306358](https://github.com/microsoft/mixed-reality-extension-sdk/pull/593#discussion_r428306358) [](ancestors = 428306358)

Reply:
I see. I misunderstood then that the point would be in the object space.